File: practice/caesar/site.js
'use strict';

/**
 * Returns the ASCII position of the inputted character.
*/
function getIndex(letter) {
  return letter.charCodeAt();
}

/**
 * Returns the ASCII character as a string associated with supplied index.
*/
function getLetter(index) {
  return String.fromCharCode(index);
}

/**
 * Takes in a string and shifts it [key] amount on the alphabet to encrypt
 * the word.
*/

function caesarEncrypt(key, inputWord) {
  var wordAsArray = inputWord.split('');
  var indexArray = _.map(wordAsArray, getIndex);
  var shiftedIndexArray = _.map(indexArray, function(index) {
    var asciiSpace = 32;
    if (index === asciiSpace) {
      return index;
    } else {
      var newIndex = index + key;
    }
    var asciiIndexZ = 122;
    var alphabetsize = 26;
    var asciiMiscCharMin = 91;
    var asciiMiscCharMax = 96;
    if (newIndex > asciiIndexZ) {
      newIndex -= alphabetsize;
    } else if (newIndex >= asciiMiscCharMin && newIndex <= asciiMiscCharMax) {
      newIndex -= alphabetsize;
    }
    return newIndex;
  });
  var outputArray = _.map(shiftedIndexArray, getLetter);
  return outputArray.join('');
}


/**
 * Takes in a string and shifts it back [key] amount on the alphabet to decrypt
 * the word.
 */

function caesarDecrypt(key, inputWord) {
  var wordAsArray = inputWord.split('');
  var indexArray = _.map(wordAsArray, getIndex);
  var shiftedIndexArray = _.map(indexArray, function addKey(index) {
    if (index === 32) {
      return index;
    } else {
      var newIndex = index - key;
    }
    if (newIndex < 65) {
      newIndex += 26;
    } else if (newIndex > 90 && newIndex < 97) {
      newIndex += 26;
    }
    return newIndex;
  });
  var outputArray = _.map(shiftedIndexArray, getLetter);
  return outputArray.join('');
}

File: practice/caesar/site.py
from string import ascii_uppercase


def get_key():
    """Returns the key for encryption as an int.

    May later contain an input function.
    """
    return 3


def get_input_word():
    """Returns the input word as an uppercase string.

    May later contain an input function.
    """
    return 'HOW ARE YOU'


def encrypt_word(key, input_word):
    i = 0
    output_list = []
    for input_letter in input_word:
        for letter in ascii_uppercase:
            if input_letter == letter:
                index = i + key
                if index > 25:
                    index -= 26
                output_list.append(ascii_uppercase[index])
            if i >= 25:
                if input_letter == ' ':
                    output_list.append(input_letter)
                i = 0
            else:
                i += 1
    return ''.join(output_list)


def caesarEncrypt():
    """Takes in a string and shifts it [key] amount on the alphabet to encrypt
    the word.
    >>> caesarEncrypt()
    'KRZ DUH BRX'
    """
    key = get_key()
    input_word = get_input_word()
    return encrypt_word(key, input_word)


def caesarDecrypt():
    """Takes in a string and shifts it back [key] amount on the alphabet to
    decrypt the word.
    """
